src/util/updateUrlHistory.js

```diff
@@ -23,7 +24,7 @@ export const updateUrlHistory = (thoughtsRanked = RANKED_ROOT, { replace, though
     return
   }
 
-  const decoded = decodeThoughtsUrl(window.location.pathname, thoughtIndex, contextIndex)
+  const decoded = decodeThoughtsUrl(store.getState(), window.location.pathname)
```

Comment:
I think we can use the same `state` instance on lines 27 and 37 instead of calling `store.getState()` twice. Please correct me if I'm wrong.